src/BaslerCamera.cpp

```diff
@@ -585,59 +478,66 @@ void Camera::_AcqThread::threadFunction()
 
         try
         {
-            WaitObjects waitset;
-            waitset.Add(m_cam.WaitObject_);
-            waitset.Add(m_cam.StreamGrabber_->GetWaitObject());
+            CGrabResultPtr ptrGrabResult;
     
             bool continueAcq = true;
             while(continueAcq && (!m_cam.m_nb_frames || m_cam.m_image_number < m_cam.m_nb_frames))
             {
 	      m_cam._setStatus(Camera::Exposure,false);
-                unsigned int event_number;
-                if(waitset.WaitForAny(m_cam.m_timeout,&event_number)) // Wait m_timeout
-                {
-                    switch(event_number)
-                    {
-                        case 0:    // event
-                            DEB_TRACE() << "Receive Event";
-                            m_cam.WaitObject_.Reset();
-                            aLock.lock();
-                            continueAcq = !m_cam.m_wait_flag && !m_cam.m_quit;
-                            aLock.unlock();
-                        break;
-                        case 1:
-                            // Get the grab result from the grabber's result queue
-                            GrabResult Result;
-                            m_cam.StreamGrabber_->RetrieveResult(Result);
-                            if (Grabbed == Result.Status())
-                            {
-                                // Grabbing was successful, process image
-                                m_cam._setStatus(Camera::Readout,false);
-                                DEB_TRACE()  << "image#" << DEB_VAR1(m_cam.m_image_number) <<" acquired !";
 				if(!m_cam.m_video_flag_mode)
 				  {
-				    int nb_buffers;
-				    buffer_mgr.getNbBuffers(nb_buffers);
-				    if (m_cam.m_nb_frames == 0 || 
-					m_cam.m_image_number < int(m_cam.m_nb_frames - nb_buffers))
-				      m_cam.StreamGrabber_->QueueBuffer(Result.Handle(),NULL);
-                                
-				    HwFrameInfoType frame_info;
-				    frame_info.acq_frame_nb = m_cam.m_image_number;
-				    // copy TmpBuffer frame to SoftBuffer frame room
-				    if (m_cam.m_nb_frames == 0)
-				      {
-					void *ptr = buffer_mgr.getFrameBufferPtr(m_cam.m_image_number);
-					memcpy(ptr, (void *)Result.Buffer(), m_cam.ImageSize_);
-				      }
-				    continueAcq = buffer_mgr.newFrameReady(frame_info);
-				    DEB_TRACE() << DEB_VAR1(continueAcq);
-				  }
+
+        // Camera.StopGrabbing() is called automatically by the RetrieveResult() method
+        // when c_countOfImagesToGrab images have been retrieved.
+        while (m_cam.Camera_->IsGrabbing())
+        {
+            // Wait for an image and then retrieve it. A timeout of 3000 ms is used.
+            m_cam.Camera_->RetrieveResult(3000, ptrGrabResult, TimeoutHandling_ThrowException);
```

Comment:
Timeout should probably be much shorter and preferably user configurable.  
It may also depend on the trigger mode.